nav2_controller/src/controller_server.cpp

```diff
@@ -63,7 +63,10 @@ ControllerServer::ControllerServer(const rclcpp::NodeOptions & options)
   // The costmap node is used in the implementation of the controller
   costmap_ros_ = std::make_shared<nav2_costmap_2d::Costmap2DROS>(
     "local_costmap", std::string{get_namespace()}, "local_costmap");
-
+  costmap_ros_->set_parameter(
```

Comment:
?

Reply:
Ah yes forgot to comment here. This is similar to what is implemented in` bt_action_server_impl.hpp`. These are nodes created internally, as opposed to in the launch file, so the SetParameter doesn't seem to have any power here.

Unfortunately I did not find a way to set the `use_sim_time` parameter during the creation of the Node here because the parent node (of the controller server) is not passed `Costmap2DROS`. The next best thing I found was to set it after its creation.

I thought it wouldn't work, I assumed the Node would need to know this parameter at creation to know which kind of clock to create/subscribe to my tests proved otherwise :shrug:

Reply:
Could you add it as a constructor argument for `Costmap2DROS` and do it in there?

Reply:
Feasible I guess but can you clarify why you like this option better? It feels a bit intrusive to me

Reply:
Also, to make sure I understand: you mean to pass only the use_sim_time parameter right? not the whole parent node? Should it default to False?

Reply:
Yeah, just add the constructor argument `bool use_sim_time` which can just be passed via `get_parameter("use_sim_time").as_bool()`. Then within the costmap we can set it when the internal node is constructed as you did in the BT action. 

Setting a parameter outside like this seems a little hacky and I think just like the name / namespace of the costmap, which clock to use seems like a pretty fundamental input. Plus, something might initialize using a wrong clock then if changes are made later if `use_sim_time` is set after construction so its protecting us from a really subtle timing bug. It seems like a bad idea to have the wrong time source initialized then being changed in the application after construction. For example, other users of Costmap2DROS outside of the application servers won't know about this change and could cause weeks of headaches trying to track down why the wrong clock is used now that the example param files don't contain that argument

Less importantly, but other yaml files in the system tests are going to have the `use_sim_time` expectation to be set that this didn't update in launch, so if that's gone / deprecated, we need to update them to have it set in code - but its easier to find if you literally change the signature to find where it needs to be done.